e2e/common/traits/health_test.go

```diff
@@ -328,5 +328,57 @@ func TestHealthTrait(t *testing.T) {
 			}))
 	})
 
+	t.Run("Startup condition with never properly started route", func(t *testing.T) {
+		name := "startup-never-ready"
+
+		Expect(KamelRunWithID(operatorID, ns, "files/NeverReady.java",
+			"-t", "health.enabled=true",
```

Comment:
I think this test is missing the equivalent of running the new probe with `-t health.startup-probe-enabled=true`